### ISSUE TITLE
[BUG] Content empty (default values) in response

### ISSUE THREAD

**Issue by bryfa** [reactiveui/refit#1202]
Hi, I am getting empty responses (or default values) on the client for the data content part. So the de-serializer is not working properly somehow. I use the Microsoft json serializer. On the controller I return a OK(Response<T>) and on the client ApiRespone<Response<T>> (or just Response<T>) where T is a serializable object or just a simple value type. Response<T> serializes and de-serializes well (I have run some tests) but through the Refit API call I will keep getting default values. I'm using version 6.0.38

**Comment by lostincomputer** [comment]
Do you have a minimal reproducible example (working source code)?

**Comment by bryfa** [comment]
`
public async Task<T> GetAuthorizedClient<T>(string? hostUrl = null)
        {
            hostUrl ??= "https://localhost:9001";
            var options = new WebApplicationFactoryClientOptions
            {
                BaseAddress = new Uri(hostUrl),
            };

            var hostApiVersionBaseAddress = $"{options.BaseAddress}api/v1";
            var httpClient  = CreateClient(options);
            
            // needed for client ip tracking
            httpClient.DefaultRequestHeaders.Add("X-Forwarded-For", new []{ "127.0.0.1" });

            var authenticationApi = RestService.For<IAuthenticationApi>(httpClient);
            var response = await authenticationApi.Authenticate(new AuthenticationRequest(EMAIL, PASSWORD));
            var authenticatedClient = CreateDefaultClient(new Uri(hostApiVersionBaseAddress),
                new AuthenticationJwtTokenHandler(response.Data!.JwtToken, new HttpLoggingHandler()));

            var settings = new RefitSettings
            {
                ContentSerializer = new SystemTextJsonContentSerializer(new JsonSerializerOptions
                {
                    Converters =
                    {
                        new EmailAddressConverter()
                    }
                }),
            };

            //return RestService.For<T>(authenticatedClient, settings);
            return RestService.For<T>(authenticatedClient);
        }

`
        
I think I have found the issue. When providing the settings then the response gets empty also because I dont have any specific serializer for the DTO class that is in the response. When I remove this settings all works perfectly. 

So how can I support both?

**Comment by lostincomputer** [comment]
I'm guessing that your `EmailAddressConverter` is incorrectly implemented. It probably is trying to deserialize all types instead of only email addresses. I unfortunately don't have enough information to verify my theory because your reply didn't include `EmailAddressConverter`.

What you are configuring is System.Text.Json, not Refit, so encourage you to refer to [System.Text.Json documentation](https://docs.microsoft.com/en-us/dotnet/api/system.text.json?view=net-5.0) and how to write [custom JsonConverters](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-converters-how-to?pivots=dotnet-5-0)

**Comment by bryfa** [comment]
The email address converter works perfectly in other contexts. I also tested this. I already did have a look at the Text.Json docss and I implemented it the way it as it is proposed. So I dont think this is the issue here.

**Comment by lostincomputer** [comment]
I'm unfortunately unable to reproduce the problem. Perhaps you could sent a working/compiling code that reproduces the problem

**Comment by bryfa** [comment]
Yes I can do that but then I need some more time.  Will prepare something later on.
Btw. the model/DTO in the first sample doesn't make use of the EmailAddress serializer/value converter.  When the settings are provided I get an empty response only.

public class EmailAddressConverter : JsonConverter<EmailAddress>
{
       public override EmailAddress? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
        {
            var value = reader.GetString()!;
            return string.IsNullOrWhiteSpace(value) ? EmailAddress.CreateEmpty() : EmailAddress.Create(value);
        }
        
        public override void Write(Utf8JsonWriter writer, EmailAddress value, JsonSerializerOptions options)
        {
            writer.WriteStringValue(value == null || value.IsEmpty() ? null : value.ToString());
        }
 }

Another question related to this. Is there a way to track issues on the serializer?
